test-network-function/certification/suite.go

```diff
@@ -68,9 +69,71 @@ var _ = ginkgo.Describe(common.AffiliatedCertTestKey, func() {
 
 		testContainerCertificationStatus()
 		testAllOperatorCertified(env)
+		testHelmCertified(env)
 	}
 })
 
+//
+/*func helmCertified() func() (interface{}, error) {
+	return func() (interface{}, error) {
+		return certAPIClient.GetYamlFile()
+	}
+}*/
+func testHelmCertified(env *configpkg.TestEnvironment) {
```

Comment:
this test would never fail.
Also, I would say the logic is a  hard to follow. there's 7 levels of fors and ifs in it.
Can you add comments or make it a little bit simpler